wpimath/src/main/native/include/frc/estimator/MerweScaledSigmaPoints.h

```diff
@@ -52,22 +51,22 @@ class MerweScaledSigmaPoints {
 
   /**
    * Computes the sigma points for an unscented Kalman filter given the mean
-   * (x) and covariance(P) of the filter.
+   * (x) and square-root covariance(S) of the filter.
    *
    * @param x An array of the means.
-   * @param P Covariance of the filter.
+   * @param S Square-root covariance of the filter.
    *
    * @return Two dimensional array of sigma points. Each column contains all of
    *         the sigmas for one dimension in the problem space. Ordered by
    *         Xi_0, Xi_{1..n}, Xi_{n+1..2n}.
    *
    */
-  Eigen::Matrix<double, States, 2 * States + 1> SigmaPoints(
+  Eigen::Matrix<double, States, 2 * States + 1> SquareRootSigmaPoints(
       const Eigen::Vector<double, States>& x,
-      const Eigen::Matrix<double, States, States>& P) {
+      const Eigen::Matrix<double, States, States>& S) {
     double lambda = std::pow(m_alpha, 2) * (States + m_kappa) - States;
-    Eigen::Matrix<double, States, States> U =
-        ((lambda + States) * P).llt().matrixL();
+    double eta = std::sqrt(lambda + States);
+    auto U = eta * S;
```

Comment:
`auto` is dangerous here because it could deduce as a temporary type. That's one of the downsides of Eigen using expression templates. Use `Eigen::Matrix<double, States, States>` instead.